php_phongo.c

```diff
@@ -1375,7 +1375,8 @@ static bool php_phongo_uri_finalize_auth(mongoc_uri_t* uri) /* {{{ */
 		}
 
 		/* MONGODB-X509 is the only mechanism that doesn't require username */
-		if (strcasecmp(mongoc_uri_get_auth_mechanism(uri), "MONGODB-X509")) {
+		if (strcasecmp(mongoc_uri_get_auth_mechanism(uri), "MONGODB-X509") &&
+			strcasecmp(mongoc_uri_get_auth_mechanism(uri), "MONGODB-AWS")) {
```

Comment:
I suppose the comment above should be updated, since MONGODB-X509 is no longer the only mechanism to not require a username.